// @bbc/igm-btn/src/Btn.js
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Btn = (props) => {
  const {
    children,
    className,
    type,
    htmlType,
    disabled,
    ...otherAttributes
  } = props;

  const types = ['link', 'plainLink', 'primary', 'secondary', 'danger', 'info', 'publish'];

  const classes = classnames({
    'igm-btn': true,
    [`igm-btn--${type}`]: types.indexOf(type) !== -1,
  }, className);

  if (children) {
    return (
      <button
        disabled={disabled}
        type={htmlType}
        className={classes}
        {...otherAttributes}
      >
        {children}
      </button>
    );
  }
  return null;
};

Btn.propTypes = {
  /** Contents of the button */
  children: PropTypes.node,
  /** Custom class added to the button */
  className: PropTypes.string,
  /** The type determines the color of the button */
  type: PropTypes.oneOf([
    'primary',
    'secondary',
    'info',
    'danger',
    'publish',
    'link',
    'plainLink',
  ]),
  /** HTML type for the button */
  htmlType: PropTypes.oneOf(['button', 'submit', 'reset']),
  /** Disable the button */
  disabled: PropTypes.bool,
};

Btn.defaultProps = {
  htmlType: 'button',
  disabled: false,
};

export default Btn;
